python/tests/unit/arcticdb/version_store/test_append.py

```diff
@@ -288,3 +288,34 @@ def test_(initial: InputFactories, append: InputFactories, match, swap, lmdb_ver
     to_append, _ = append.make(abs(next_start), 1)
     with pytest.raises(NormalizationException):
         lib.append("s", to_append, match=match)
+
+
```

Comment:
This is not a small change, so really need a lot more tests for each. I would expect:

* `append(ensure_sorted=False)`
* `RangeIndex`
* `MultiIndex`
* Various combinations of ascending, descending and unsorted in both existing data and the argument (you might want to look into Hypothesis)